var/spack/repos/builtin/packages/root/package.py

```diff
@@ -12,34 +12,38 @@ class Root(CMakePackage):
     """ROOT is a data analysis framework."""
 
     homepage = "https://root.cern.ch"
-    url      = "https://root.cern/download/root_v6.14.00.source.tar.gz"
+    url      = "https://root.cern/download/root_v6.16.00.source.tar.gz"
 
     # ###################### Versions ##########################
 
     # Master branch
     version('master', git="https://github.com/root-project/root.git",
         branch='master')
 
-    # Development version
-    version('6.15.02', sha256='2236fe4935139459239c935b2f93b3aa6bbfe92765ae9d9db9dd0b947bf19071')
+    # Development version (when more recent than production).
 
     # Production version
     version('6.16.00', sha256='2a45055c6091adaa72b977c512f84da8ef92723c30837c7e2643eecc9c5ce4d8', preferred=True)
 
     # Old versions
+    version('6.14.08', sha256='1b63b51cfb4dc20f1f5749faac6bbd1098eccab777f8b49911257d77186c73c4')
+    version('6.14.06', sha256='0fb943b61396f282b289e35c455a9ab60126229be1bd3f04a8f00b37c13ab432')
     version('6.14.04', sha256='463ec20692332a422cfb5f38c78bedab1c40ab4d81be18e99b50cf9f53f596cf')
     version('6.14.02', sha256='93816519523e87ac75924178d87112d1573eaa108fc65691aea9a9dd5bc05b3e')
     version('6.14.00', sha256='7946430373489310c2791ff7a3520e393dc059db1371272bcd9d9cf0df347a0b')
-    version('6.13.08', sha256='353e6aab4826b981190ad45678cc7d3d9ca5ba02469e5d8aa46543ea61961139')
     version('6.12.06', sha256='aedcfd2257806e425b9f61b483e25ba600eb0ea606e21262eafaa9dc745aa794')
     version('6.10.08', sha256='2cd276d2ac365403c66f08edd1be62fe932a0334f76349b24d8c737c0d6dad8a')
-    version('6.09.02', sha256='5348096084adea514297050884baa33f6cf6fd9e91e83e9967c6f07528588639')
     version('6.08.06', sha256='ea31b047ba6fc04b0b312667349eaf1498a254ccacd212144f15ffcb3f5c0592')
     version('6.06.08', sha256='7cb836282014cce822ef589cad27811eb7a86d7fad45a871fa6b0e6319ec201a')
     version('6.06.06', sha256='0a7d702a130a260c72cb6ea754359eaee49a8c4531b31f23de0bfcafe3ce466b')
     version('6.06.04', sha256='ab86dcc80cbd8e704099af0789e23f49469932ac4936d2291602301a7aa8795b')
     version('6.06.02', sha256='18a4ce42ee19e1a810d5351f74ec9550e6e422b13b5c58e0c3db740cdbc569d1')
-    version('5.34.36', sha256='fc868e5f4905544c3f392cc9e895ef5571a08e48682e7fe173bd44c0ba0c7dcd')
+    version('5.34.38', sha256='2c3bda69601d94836bdd88283a6585b4774eafc813deb6aa348df0af2922c4d2')
+
+    patch('format-stringbuf-size.patch', level=0)
+    patch('find-mysql.patch', level=1)
+    patch('honor-unuran-switch.patch', level=1, when='@:6.13.99')
+    patch('root7-webgui.patch', level=1, when='@6.16.00')
```

Comment:
Should this patch be reported to the ROOT team?

Reply:
The ROOT team are aware of the first two patches; the latter two have been incorporated into recent releases. I will re-up the discussion with the ROOT maintainers on the first two, but the stringbuf patch has been used by FNAL and CMS for years without ROOT being interested AFAICT.

Reply:
The MySQL patch is a PR root-project/ROOT#1993 which (I discovered this morning) I'd allowed to become moribund. I will revive it.

Reply:
The stringbuf patch appears to be a performance-based patch when a significant proportion of strings are larger than the original optimization would cater for, but not so large that the slower method would be better. The patch triples the size of two thread-local static buffers. from 4KB and 2KB to 12KB and 6KB respectively.

Reply:
@chissg for all patches, could you add a comment pointing to the relevant link (assuming you did not write them yourself, in which case could you add details about why it was needed)?

Reply:
@scheibelp see https://github.com/spack/spack/pull/11215/commits/2b38577f800c52b58d6ad0a47708b1b80819d07c.

Reply:
Thank you for the clarification @chissg